presto-common/src/main/java/com/facebook/presto/common/block/BlockBuilderStatus.java

```diff
@@ -74,7 +74,7 @@ private static int deepInstanceSize(Class<?> clazz)
 
         int size = ClassLayout.parseClass(clazz).instanceSize();
         for (Field field : clazz.getDeclaredFields()) {
-            if (!field.getType().isPrimitive()) {
+            if (!field.getType().isPrimitive() && !field.isSynthetic()) {
                 size += deepInstanceSize(field.getType());
```

Comment:
Can we add a comment here to indicate `isSynthetic` is only used for code coverage?